formats/hocon/src/main/kotlin/kotlinx/serialization/hocon/HoconDecoder.kt

```diff
@@ -0,0 +1,49 @@
+package kotlinx.serialization.hocon
+
+import com.typesafe.config.Config
+import kotlinx.serialization.ExperimentalSerializationApi
+
+/**
+ * Decoder used by Hocon during deserialization.
+ * This interface allows to call methods from the Lightbend/config library on the [Config] object.
+ *
+ * Usage example (nested config serialization):
+ * ```
+ * @Serializable
+ * data class Example(
+ *   @Serializable(NestedConfigSerializer::class)
+ *   val d: Config
+ * )
+ *
+ * object NestedConfigSerializer : KSerializer<Config> {
+ *     override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("package.Config", PrimitiveKind.STRING)
+ *
+ *     override fun deserialize(decoder: Decoder): Config = if (decoder is HoconDecoder) {
```

Comment:
See my comment for ConfigMemorySizeSerializer

Reply:
Fixed